Mathlib/Topology/MetricSpace/Congruence.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2024 Jovan Gerbscheid. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jovan Gerbscheid, Newell Jensen
+-/
+import Mathlib.Topology.MetricSpace.Isometry
+
+/-!
+# Congruences
+
+In this file we define congruence, i.e., the equivalence between sets of points in
+a metric space where all corresponding pairwise distances are the same. The motivating
+example are triangles in the plane.
+
+## Main results
+
+In the case of an `EMetricSpace` we show an `IsometryEquiv` between the points:
+`Set.range v₁ ≃ᵢ Set.range v₂`.
+
+## Notation
+
+* `v₁ ≅ v₂`: for `congruence v₁ v₂`.
+-/
+
+variable {ι ι' : Type*} {P₁ P₂ P₃ : Type*} {v₁ : ι → P₁} {v₂ : ι → P₂} {v₃ : ι → P₃}
+
+noncomputable section
+
+/-- A congruence between indexed sets of vertices v₁ and v₂.
+Use `open scoped Congruence` to access the `v₁ ≅ v₂` notation. -/
```

Comment:
Does it mean that we can't deal with congruence and category theory at the same time?

Reply:
I don't know as I haven't done much with category theory in mathlib.  Maybe @jsm28 knows?

Reply:
I have no idea what the effects of this notation would be on using congruence together with category theory. (I do expect to have a use for category theory in *combinatorial* geometry - it's how mathlib handles compactness arguments - though probably not for congruence of families of points in that combinatorial geometry.)